src/providers/tokens.provider.ts

```diff
@@ -198,7 +203,7 @@ export default {
       isLoading: priceQueryLoading,
       isError: priceQueryError,
       refetch: refetchPrices
-    } = useTokenPricesQuery(tokenAddresses);
+    } = useTokenPricesQuery(tokenAddresses, injectedPrices);
```

Comment:
Like we do for the allowance contracts.
```suggestion
    } = useTokenPricesQuery(tokenAddresses, toRef(state, 'injectedPrices'));
```